packages/file-search/src/browser/quick-file-open.ts

```diff
@@ -126,79 +130,76 @@ export class QuickFileOpenService implements QuickOpenModel {
-            if (!token.isCancellationRequested) {
-                const root = new URI(rootUri);
-                result.forEach(p => {
-                    const uri = root.withPath(root.path.join(p)).toString();
-                    proposed.add(uri);
-                });
-                const itemPromises = Array.from(proposed).map(uri => this.toItem(uri));
-                acceptor(await Promise.all(itemPromises));
+        const root = new URI(rootUri);
+        const alreadyCollected = new Set<string>();
+        
+        const recentlyUsedItems: QuickOpenItem[] = [];
+        const locations = this.navigationLocationservice.locations()
+        if (locations.length > 0) {
+            for (const location of locations) {
+                const uriString = location.uri.toString();
+                if (!alreadyCollected.has(uriString) && fuzzy.test(lookFor, uriString)) {
+                    recentlyUsedItems.push(await this.toItem(location.uri, recentlyUsedItems.length === 0 ? 'recently used' : undefined));
+                    alreadyCollected.add(uriString);
+                }
             }
-        };
-        this.fileSearchService.find(lookFor, {
-            rootUri,
-            fuzzyMatch: true,
-            limit: 200,
-            useGitIgnore: this.hideIgnoredFiles,
-        }, token).then(handler);
-    }
-
-    private async toItem(uriString: string) {
-        const uri = new URI(uriString);
-        return new FileQuickOpenItem(uri,
-            this.labelProvider.getName(uri),
-            await this.labelProvider.getIcon(uri),
-            this.labelProvider.getLongName(uri.parent),
-            this.openerService);
-    }
-
-}
-
-export class FileQuickOpenItem extends QuickOpenItem {
-
-    constructor(
-        protected readonly uri: URI,
-        protected readonly label: string,
-        protected readonly icon: string,
-        protected readonly parent: string,
-        protected readonly openerService: OpenerService
-    ) {
-        super();
-    }
-
-    getLabel(): string {
-        return this.label;
-    }
-
-    isHidden(): boolean {
-        return false;
-    }
-
-    getTooltip(): string {
-        return this.uri.path.toString();
-    }
-
-    getDescription(): string {
-        return this.parent;
-    }
-
-    getUri(): URI {
-        return this.uri;
+        }
+        if (lookFor.length > 0) {    
+            const handler = async (result: string[]) => {
+                if (!token.isCancellationRequested) {
+                    const fileSearchResultItems: QuickOpenItem[] = [];
+                    for (const p of result) {
+                        const uri = root.withPath(root.path.join(p));
+                        const uriString = uri.toString();
+                        if (!alreadyCollected.has(uriString)) {
+                            fileSearchResultItems.push(await this.toItem(uri, fileSearchResultItems.length === 0 ? 'search result' : undefined));
```

Comment:
Note: VS Code says: `file results`.

Reply:
changed it, although search result makes more sense to me :)